backend/controllers/bulkAppointmentsClashes.js

```diff
@@ -82,6 +82,17 @@ const bulkAppointmentsClashesFactory = (elite2Api, logError) => {
     } = req.session
 
     const remainingPrisoners = prisonersListed.filter(prisoner => !req.body[prisoner.offenderNo])
+    const removedPrisoners = prisonersListed.filter(prisoner => req.body[prisoner.offenderNo])
```

Comment:
Could probably use .reduce and go through `prisonersListed` once.

Reply:
I was thinking something like...
```
const [removedPrisoners, remainingPrisoners] = prisonersListed.reduce(
      (prisonersSeparated, prisoner) => {
        prisonersSeparated[req.body[prisoner.offenderNo] ? 0 : 1].push(prisoner)
        return prisonersSeparated
      },
      [[], []]
    )
```

Happy to discuss tomorrow though.

Reply:
Nice, I'd done this:

`    const prisonersRemoved = prisonersListed.reduce((removedList, prisoner) => {
      req.body[prisoner.offenderNo] && removedList.push({
        offenderNo: prisoner.offenderNo,
        lastName: prisoner.lastName
      });
      return removedList;
    }, [])`

But that caters for both nicely.